### DISCUSSION TITLE
Android Integration

### DISCUSSION THREAD

**Discussion by ghost** [tbroyer/gradle-errorprone-plugin#80] [Q&A] [answered]
Hi,

I am trying to integrate error prone into an Android SDK. Error prone don't have an official Gradle support and recommends this plugin instead. However, Android support is removed in 3.0+ according to the release notes. Release notes also states that the plugin should still be usable after a manual configuration of Java compile task. 

The closest thing I could find on how to do this is here https://github.com/tbroyer/gradle-errorprone-plugin/issues/26#issuecomment-489086884. The solution provided works for me but I am a bit worried that the compiler argument (-XDcompilePolicy=simple) might cause issues at runtime and not sure if it is safe to use as is. I am trying to decide whether to proceed with the solution in the comment or not integrate error prone at all.

Thank you.

**Comment by tbroyer** [comment, marked as answer]
From the README:
* “This plugin creates a configuration named `errorprone`, and configures the `<sourceSet>AnnotationProcessor` configuration for each source set to extend it.” → the source sets with AGP aren't the same, so no configuration is configured to extend the `errorprone` configuration
* “Note that this plugin only enables Error Prone on tasks for source sets” → again, no source set so Error Prone is disabled

So you'd want to follow the [“If you're creating custom compile tasks, then you'll have to configure them manually to enable Error Prone”](https://github.com/tbroyer/gradle-errorprone-plugin#:~:text=If%20you%27re%20creating%20custom%20compile%20tasks%2C%20then%20you%27ll%20have%20to%20configure%20them%20manually%20to%20enable%20Error%20Prone) instructions then, or equivalent (AGP creates its own JavaCompile tasks, unrelated to source sets, so considered "custom" here). That's what is meant by "manual configuration" in the 3.0 release notes.

Wrt the `-XDcompilePolicy=simple` compiler argument, that's what the plugin configures anyway; but it shouldn't have any impact on the produced classes (you can verify this by compiling with and without that argument or Error Prone: the `.class` files should be exactly the same).

Fwiw, the reason that Android support was removed from the plugin is that AGP actively makes it hard to configure tasks directly, and get the tasks associated with each variant. As I don't build for Android myself, and more and more Android projects use Kotlin anyway, trying to maintain that best-effort support wasn't a good use of my time.

**Reply by ghost** [reply]
I've experimented with the compile policy flag and actually found out that it cause changes in the produced class files. They are probably only related to optimization and won't cause any runtime problems but I want to be safe so decided to not integrate error prone at all. Anyways, thanks @tbroyer.
